rotkehlchen/history/events/structures/asset_movement.py

```diff
@@ -0,0 +1,200 @@
+import logging
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.mixins.event import AccountingEventType
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.location_details import get_formatted_location_name
+from rotkehlchen.exchanges.data_structures import hash_id
+from rotkehlchen.history.events.structures.base import (
+    HISTORY_EVENT_DB_TUPLE_WRITE,
+    HistoryBaseEntry,
+    HistoryBaseEntryType,
+)
+from rotkehlchen.history.events.structures.types import (
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval
+from rotkehlchen.types import (
+    Location,
+    TimestampMS,
+)
+from rotkehlchen.utils.misc import ts_ms_to_sec
+
+if TYPE_CHECKING:
+    from more_itertools import peekable
+
+    from rotkehlchen.accounting.mixins.event import AccountingEventMixin
+    from rotkehlchen.accounting.pot import AccountingPot
+    from rotkehlchen.fval import FVal
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class AssetMovement(HistoryBaseEntry):
+    """Asset movement event representing deposits and withdrawals on exchanges."""
+
+    def __init__(
+            self,
+            sequence_index: int,
+            timestamp: TimestampMS,
+            location: Location,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
```

Comment:
In a similar fashion here, shouldn't it just be three options here? Which would translate to specific type/subtypes? Deposit, withdrawal or fee of any of the two?


I mean it's not bad to have them as arguments. But depending on where you will use this there will need to be some logic on what to use. So it coudl all be in the constructor and simplify the amount of arguments.

Really depends on where/how this is used.
```suggestion
```

Reply:
Yes, we only have deposit/withdrawal/fee right?

Reply:
Yes, should only ever be deposit, deposit fee, withdrawal, withdrawal fee here. I've changed it to use AssetMovementCategory which was used in the old AssetMovement class, with added values for deposit fees and withdrawal fees.